jdaviz/core/launcher.py

```diff
@@ -74,59 +82,99 @@ def _launch_config_with_data(config, data=None, show=True, **kwargs):
         else:
-            compatible_helpers, loaded_data = identify_helper(filepath)
-            if len(compatible_helpers) > 0 and loaded_data is None:
-                loaded_data = filepath
-
-        for config, btn in btns.items():
-            btn.disabled = not (config in compatible_helpers)
-
-    id_data_btn = v.Btn(class_="ma-2", outlined=True, color="primary",
-                        children=[v.Icon(children=["mdi-magnify"])])
-    id_data_btn.on_event('click', lambda btn, event, data: enable_compatible_configs(btn.value))
-    jslink((text_field, 'v_model'), (id_data_btn, 'value'))
-
-    filepath_row.children = [text_field, id_data_btn]
-
-    # Create Launcher
-    main.children = [intro_row, filepath_row, btn_row]
+            path = Path(self.filepath)
+            if not path.is_file():
+                self.loaded_data = None
+                self.hint = "Cannot find file on disk. Please check your filepath"
+            else:
+                try:
+                    self.compatible_configs, self.loaded_data = identify_helper(self.filepath)
+                    self.hint = "Compatible configs identified! Please select your config:"
+                except Exception:
+                    self.hint = "Could not autoidentify compatible config. Please manually select config to load your data into:"
+                    self.compatible_configs = self.configs
+                    self.loaded_data = self.filepath
+                finally:
+                    if len(self.compatible_configs) > 0 and self.loaded_data is None:
+                        self.loaded_data = self.filepath
+        # Clear hint if it's still stuck on "Identifying". We're in an ambiguous state
+        self.hint = '' if self.hint == "Please wait. Identifying file..." else self.hint
+
+    def vue_choose_file(self, *args, **kwargs):
+        if self._file_chooser.file_path is None:
+            self.error_message = "No file selected"
+        elif Path(self._file_chooser.file_path).is_file():
+            self.file_chooser_visible = False
+            self.filepath = self._file_chooser.file_path
+
+    def vue_launch_config(self, config):
+        helper = _launch_config_with_data(config, self.loaded_data, show=False)
+        helper.app.layout.height = self.height
+        helper.app.state.settings['context']['notebook']['max_height'] = self.height
+        self.main.color = 'transparent'
+        self.main.children = [helper.app]
+        
+
+def show_launcher(configs=ALL_JDAVIZ_CONFIGS, height="100%"):
+    # Color defined manually due to the custom theme not being defined yet (in app.vue)
+    height = f"{height}px" if isinstance(height, int) else height
+    main = v.Sheet(class_="mx-25",
+                   color="#00212C",
+                   height=height,
+                   _metadata={'mount_id': 'content'})
+    main.children = [Launcher(main, configs, height)]
 
     return main
```

Comment:
maybe unrelated to this PR, but I'm wondering if this should call `display` instead of just returning to be consistent with other `show` methods (otherwise if this isn't the last line in a cell, it won't display).